### PR TITLE
OIDC integration & dependency upgrade

### PR DESCRIPTION
**Pull request by NShani**
## Purpose
Purpose of this implementation to support authentication and authorization to scan manger ( which is build for doing the security scans) by integrating with WSO2 IS to support OIDC.

## Goals
Provide the authentication to scan manager and monitor the scans along with the user details.

## Approach
Integrating with Identity Server and supporting OIDC authentication to Scan Manager and keep the user details in the session.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: internal/scan-manager/webapp/src/main/java/org/wso2/security/tools/scanmanager/webapp/controller/ScanController.java [hunk 1]
```diff
@@ -118,8 +120,15 @@ public String startScan(MultipartHttpServletRequest multipartHttpServletRequest)
                 parameterMap.put(requestParamMap.getKey(), requestParamMap.getValue()[0]);
             }
         }
-        String username = (String) (((StandardMultipartHttpServletRequest) multipartHttpServletRequest).getRequest())
-                .getSession().getAttribute(SCAN_REQUEST_USER_NAME);
+        String username = null;
+        Authentication authentication = SecurityContextHolder.getContext()
```
Comment (dulanjal) [inline internal/scan-manager/webapp/src/main/java/org/wso2/security/tools/scanmanager/webapp/controller/ScanController.java:124]: Can `authentication` become null?

File: internal/scan-manager/webapp/src/main/java/org/wso2/security/tools/scanmanager/webapp/service/ScanServiceImpl.java [hunk 2]
```diff
@@ -293,92 +381,4 @@ private ResponseEntity sendSubmitScanRequest(Map<String, String> uploadedFileMap
...
-    public boolean clearScan(String id) {
-        // Clear scan from preparing scans list if the scan status is ERROR.
-        if (preparingScans.containsKey(id) && preparingScans.get(id).getStatus() == ScanStatus.ERROR) {
-            preparingScans.remove(id);
-            return logService.removeLogsForPreparingScan(id);
-        } else {
-            return false;
-        }
-    }
 }
```
Comment (dulanjal) [inline internal/scan-manager/webapp/src/main/java/org/wso2/security/tools/scanmanager/webapp/service/ScanServiceImpl.java:384]: What has changed here?

File: internal/scan-manager/webapp/src/main/java/org/wso2/security/tools/scanmanager/webapp/service/ScanServiceImpl.java [hunk 3]
```diff
@@ -170,6 +170,94 @@ public Scan submitScan(Map<String, MultipartFile> fileMap, Map<String, String> p
...
+    public boolean clearScan(String id) {
+        // Clear scan from preparing scans list if the scan status is ERROR.
+        if (preparingScans.containsKey(id) && preparingScans.get(id).getStatus() == ScanStatus.ERROR) {
+            preparingScans.remove(id);
+            return logService.removeLogsForPreparingScan(id);
+        } else {
+            return false;
+        }
+    }
+
```
Comment (dulanjal) [inline internal/scan-manager/webapp/src/main/java/org/wso2/security/tools/scanmanager/webapp/service/ScanServiceImpl.java:260]: What has changed here?

**Comment by NShani** [comment]
> 1. From the diff, it is hard to identify the actual change in ScanServiceImpl.java.
Here, only the formatting of the code is changed.
> 2. Also, don't we have to delete org.wso2.security.tools.scanmanager.webapp.filter package? AFAIS, there was only one class in it, and now that class is deleted. If that package is already deleted, please ignore this comment.
org.wso2.security.tools.scanmanager.webapp.filter package is deleted.

File: internal/scan-manager/webapp/src/main/java/org/wso2/security/tools/scanmanager/webapp/service/ScanServiceImpl.java (hunk 3 above)
Comment (NShani) [reply internal/scan-manager/webapp/src/main/java/org/wso2/security/tools/scanmanager/webapp/service/ScanServiceImpl.java:260]: Only the code formatting is changed.

File: internal/scan-manager/webapp/src/main/java/org/wso2/security/tools/scanmanager/webapp/controller/ScanController.java (hunk 1 above)
Comment (NShani) [reply internal/scan-manager/webapp/src/main/java/org/wso2/security/tools/scanmanager/webapp/controller/ScanController.java:124]: No. If that value is null, scan manager redirects it to Identity Provider to login. Therefore when it comes to this line, always authentication has a value.

**Comment by ayomawdb** [comment]
Travis build has passed https://travis-ci.org/github/wso2/security-tools/builds/674509742

**Pull request merged**